packages/cubejs-schema-compiler/src/parser/SqlParser.ts

```diff
@@ -64,20 +64,45 @@ export class SqlParser {
   protected static sqlUpperCase(sql) {
     let result = '';
     let openChar;
+    let inComment = false;
+    let commentType = '';
 
     for (let i = 0; i < sql.length; i++) {
-      if (openChar) {
+      if (inComment) {
+        if (commentType === '--' && (sql[i] === '\n' || sql[i] === '\r')) {
```

Comment:
And it looks like that you can use only `commentType` as discriminator. Having `commentType === ''` means that you're not in comment. So you don't need `inComment` var at all.